Assets/Fungus/Thirdparty/FungusLua/Scripts/ConversationManager.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using UnityEngine;
+
+namespace Fungus
+{
+	public class ConversationManager
+	{
+		protected Stage stage;
+
+		protected CharacterItem[] characters;
+
+		protected struct CharacterItem
+		{
+			public string name;
+			public string nameText;
+			public Character character;
+
+			public bool CharacterMatch(string matchString)
+			{
+				return name.StartsWith(matchString, true, System.Globalization.CultureInfo.CurrentCulture)
+					|| nameText.StartsWith(matchString, true, System.Globalization.CultureInfo.CurrentCulture);
+			}
+		}
+
+		protected bool exitSayWait;
+
+		public ConversationManager () {
+			stage = Stage.activeStages[0];
+		
+			Character[] characterObjects = GameObject.FindObjectsOfType<Fungus.Character>();
+			characters = new CharacterItem[characterObjects.Length];
+
+			for (int i = 0; i < characterObjects.Length; i++)
+			{
+				characters[i].name = characterObjects[i].name;
+				characters[i].nameText = characterObjects[i].nameText;
+				characters[i].character = characterObjects[i];
+			}
+		}
+
+		/// <summary>
+		/// Parse and execute a conversation string
+		/// </summary>
+		/// <param name="conv"></param>
+		public IEnumerator Conversation(string conv)
+		{
+			if (string.IsNullOrEmpty(conv))
+			{
+				yield break;
+			}
+			
+			SayDialog sayDialog = SayDialog.GetSayDialog();
+
+			if (sayDialog == null)
+			{
+				yield break;
+			}
+
+			//find SimpleScript say strings with portrait options
+			//You can test regex matches here: http://regexstorm.net/tester
+			var sayRegex = new Regex(@"((?<character>\w*)( ?(?<portrait>\w*)( ?(?<position>\w*)))?:)?(?<text>.*\r*\n)");
```

Comment:
What happens if there's a space in the name of a character, portrait or position? I see some code in Stage.cs to handle this for positions alright.

Some ideas: 
1. Strings with spaces must be enclosed with quotation marks. Complicates the parser code, but it's an easy to understand rule.
2. Strip spaces from names, or replace with underscores. Easier to implement, maybe a bit harder to understand for users. 
3. Use commas as a delimiter. Split on commas and trim each string. Could be an optional format to support names containing spaces.